tools/src/test/resources/dev/run-qualification-tests.sh

```diff
@@ -0,0 +1,395 @@
+   log_msg
+   log_msg "Example Usage:"
+   log_msg "  run-qualification-tests.sh --cp=\$CLASS_PATH --heap=5g"
+   log_msg "  This is equivalent to:"
+   log_msg "    java -Xmx5g \\"
+   log_msg "         -cp rapids-4-spark-tools_2.12-<version>-SNAPSHOT.jar:\$CLASS_PATH \\"
+   log_msg "         com.nvidia.spark.rapids.tool.qualification.QualificationMain \\"
+   log_msg "         --no-html-report --output-directory file:qualification-output/csv \\"
+   log_msg "         \$LOGFILES"
+   log_msg
+   log_msg "How to Add New Test:"
+   log_msg " 1- add a dummy csv file in qual-expectations-dir <new_unit_test.csv>"
+   log_msg " 2- add the eventlog into the prof-log-dir <new_unit_log>"
+   log_msg " 3- update the definition of the hash in define_qualification_tests_map() to map between"
+   log_msg "    the expected file and the unit test name."
+   log_msg " 4- run teh script. It is expected that the script fails because the dummy csv is incorrect."
+   log_msg " 5- copy the content of the generated output dev/qualification-output/csv/new_unit_test.csv"
+   log_msg "    into the expectation folders."
+   log_msg
+}
+
+log_error()
+{
+  log_msg "${RED}$*"
+}
+
+log_result()
+{
+  log_msg "${GREEN}$*"
+}
+
+log_info() {
+  log_msg "${YELLOW}$*"
+}
+
+log_msg()
+{
+  echo -e "$* $ENDCOLOR"
+}
+
+print_banner()
+{
+  printf '%80s\n' | tr ' ' -
+  log_info "$*"
+  printf '%80s\n' | tr ' ' -
+}
+
+set_rapids_jars_from_work_dir()
+{
+  rapids_tools_jar_file=( "$( find "${MODULE_PATH}" -type f \( -iname "*.jar" ! -iname "*tests.jar" ! -iname "original-rapids-4*.jar" \) )" )
+  # get the parent directory
+  rapids_jar_home="$(dirname "${rapids_tools_jar_file}")"
+}
+
+set_rapids_jar_home()
+{
+  if [ "${rapids_jar_home}" ]; then
+    log_msg "RAPIDS_JARS are passed through the arguments: ${rapids_jar_home}/*"
+  else
+    set_rapids_jars_from_work_dir
+    log_msg "RAPIDS_JARS are pulled from : ${rapids_jar_home}/*"
+  fi
+}
+
+set_rapids_tools_classpath()
+{
+  tools_cp="${rapids_jar_home}/*"
+  log_msg "tools_cp:\n ${tools_cp}"
+  # check that the java class path is set correctly
+  if [ "$java_cp" ]; then
+    RAPIDS_CLASS_PATH="${tools_cp}:${java_cp}"
+  else
+    log_msg "The run did not define CP Dependencies"
+    RAPIDS_CLASS_PATH="${tools_cp}"
+  fi
+
+  log_msg "RAPIDS CLASS_PATH is:\n ${RAPIDS_CLASS_PATH}"
+}
+
+set_script_output()
+{
+  if [ "${script_out_dir}" ]; then
+    log_msg "Script output is set through the arguments: ${script_out_dir}"
+    # reset default values
+    # Note that it is safer to use the output-directory as a parent. This avoids the mistake of
+    # deleting user's data if the directory has other subdirectories.
+    BATCH_OUT_DIR="${script_out_dir}/qualification-output"
+    CSV_OUT_DIR="${BATCH_OUT_DIR}/csv"
+    RUNS_OUT_DIR="${BATCH_OUT_DIR}/runs-output"
+  else
+    log_msg "Script output is set as default: ${BATCH_OUT_DIR}"
+  fi
+
+  # cleanup output folder before running tests
+  if [ -d "$BATCH_OUT_DIR" ]; then
```

Comment:
it would be safer to fail if directory exists unless users adds and overwrite flag

Reply:
I thought about that, but the problem with giving the user option not to overwrite is that a csv from previous run that was not generated by the most recent run would cause the final directory comparison to pass successfully.

Reply:
I'm not sure I understand, it would fail if any previous run is there and you try to run it again.  If you select overwrite then it removes it first.
I'm ok with leaving it this way, but please add comment in top level documentation.

Reply:
Good point. I added one more flag "-f|--force"